tardis/montecarlo/montecarlo_numba/nonhomologous_grid.py

```diff
@@ -0,0 +1,308 @@
+import numpy as np
+from numba import njit
+
+from tardis.montecarlo.montecarlo_numba import (
+    njit_dict_no_parallel,
+)
+
+
+@njit(**njit_dict_no_parallel)
+def velocity(r_packet, numba_model):
+    """
+    Velocity at radius r
+
+    Parameters
+    ----------
+    r_packet: RPacket
+    numba_model: NumbaModel
+
+    Returns
+    -----------
+    v: float, current velocity
+    """
+    shell_id = r_packet.current_shell_id
+    v_inner = numba_model.v_inner[shell_id]
+    v_outer = numba_model.v_outer[shell_id]
+    r_inner = numba_model.r_inner[shell_id]
+    r_outer = numba_model.r_outer[shell_id]
+    r = r_packet.r
+    frac = (v_outer - v_inner) / (r_outer - r_inner)
+    return v_inner + frac * (r - r_inner)
+
+
+@njit(**njit_dict_no_parallel)
+def dv_dr(r_packet, numba_model):
+    """
+    dv/dr for the current shell
+
+    Parameters
+    ----------
+    r_packet: RPacket
+    numba_model: NumbaModel
+
+    Returns
+    -----------
+    dvdr: float, dv/dr of the current shell
+    """
+    shell_id = r_packet.current_shell_id
+    v_inner = numba_model.v_inner[shell_id]
+    v_outer = numba_model.v_outer[shell_id]
+    r_inner = numba_model.r_inner[shell_id]
+    r_outer = numba_model.r_outer[shell_id]
+    return (v_outer - v_inner) / (r_outer - r_inner)
```

Comment:
This is already computed by the "velocity" function. You can just return `frac` from `velocity` and get both at once.

Reply:
should I rename "velocity" function then?

Reply:
You could call it `velocity_dvdr` perhaps